docs: add subtopics for internationalization

josmar-crwdstffng opened the pull request:
Add subtopics for internationalization guide. Update links.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

The Internationalization guide is a single topic.

Issue Number: N/A

## What is the new behavior?

The Internationalization guide is a multiple topics.

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

The links embedded in existing code comments point new subtopics.

## Other information

mary-poppins commented:
You can preview 111b447 at https://pr43245-111b447.ngbuilds.io/.

mary-poppins commented:
You can preview 88832fb at https://pr43245-88832fb.ngbuilds.io/.

mary-poppins commented:
You can preview a2265eb at https://pr43245-a2265eb.ngbuilds.io/.

mary-poppins commented:
You can preview f58efe1 at https://pr43245-f58efe1.ngbuilds.io/.

mary-poppins commented:
You can preview dc4fe47 at https://pr43245-dc4fe47.ngbuilds.io/.

petebacondarwin commented on aio/content/guide/angular-compiler-options.md:
```diff
@@ -4,21 +4,24 @@ When you use [AOT compilation](guide/aot-compiler), you can control how your app
-        "preserveWhitespaces": true,
-                  ...
-      }
+<code-example language="json">
```
The changes to this file (and others below), other than updating the links to i18n pages, do not seem relevant to this PR. Perhaps put these additional changes into another PR, or at least a separate commit?

josmar-crwdstffng replied:
The markdown was cleaned as part of the update to the link to the i18n content.

petebacondarwin reviewed (changes requested):
The separation of the topics looks good! This PR is probably blocked on the one that pulls the examples out into separate files, since the two PRs conflict with each other.

---

Removing `aio/content/guide/i18n.md` will cause lots of inbound links (from 3rd party sites) pointing to a 404 page - e.g. https://pr43245-dc4fe47.ngbuilds.io/guide/i18n.

Perhaps instead, keep this page but use the content you have put in `aio/content/guide/i18n-overview.md` instead? This would also help with SEO for this page.

Alternatively, I think we would need to setup a redirect from the `guide/i18n` to `guide/i18n-overview`.

---

Could you factor out the non-i18n changes from this PR (or at least create a separate commit)? It makes it harder to review the differences specific to this change and also makes the git history less clear.

mary-poppins commented:
You can preview 9ca9df4 at https://pr43245-9ca9df4.ngbuilds.io/.

petebacondarwin commented on aio/tests/deployment/shared/URLS_TO_REDIRECT.txt:
```diff
@@ -202,6 +202,7 @@
 /guide/change-log --> https://github.com/angular/angular/blob/master/CHANGELOG.md
 /guide/cli-quickstart --> /start
 /guide/displaying-data --> /start#template-syntax
+/guide/i18n --> /guide/i18n-overview
```
👍

josmar-crwdstffng committed:
docs: add subtopics for internationalization

Add subtopics for internationalization guide.

josmar-crwdstffng committed:
docs: add subtopics for internationalization

Add subtopics for internationalization guide.

josmar-crwdstffng committed:
docs: add subtopics for internationalization to toc

Add subtopics for internationalization guide to toc.

josmar-crwdstffng committed:
docs: add redirect for internationalization

Add redirect for internationalization guide. Add test for redirect.

josmar-crwdstffng committed:
docs: add subtopics for internationalization to PR

Add subtopics for internationalization guide to PR.

josmar-crwdstffng committed:
docs: add new internationalization links

Update link for subtopics of internationalization guide. Clean up markdown.

josmar-crwdstffng committed:
docs: add new internationalization links

Update link for subtopics of internationalization guide.

mary-poppins commented:
You can preview 5221837 at https://pr43245-5221837.ngbuilds.io/.

TeriGlover commented on aio/content/navigation.json:
```diff
@@ -371,9 +371,92 @@
+                {
+                  "url": "guide/i18n-common-merge",
+                  "title": "Merge translations into the app",
+                  "tooltip": "Mearge translations into your app"
```
```suggestion
                  "tooltip": "Merge translations into your app"
```

TeriGlover reviewed (approved):
Editing review completed with several comments. Otherwise, LGTM.

josmar-crwdstffng committed:
docs: update spelling

Update spelling per review.

Co-authored-by: TeriGlover <teriglover@google.com>

mary-poppins commented:
You can preview 21958a0 at https://pr43245-21958a0.ngbuilds.io/.

josmar-crwdstffng committed:
docs: update content

Update content per review.

josmar-crwdstffng committed:
docs: update grammar

Update grammar per review.

Co-authored-by: TeriGlover <teriglover@google.com>

josmar-crwdstffng committed:
docs: update punctuation

Update punctuation per review.

Co-authored-by: TeriGlover <teriglover@google.com>

josmar-crwdstffng committed:
docs: update content

Update content per review.

josmar-crwdstffng commented on aio/content/guide/i18n-common-merge.md:
```diff
@@ -0,0 +1,264 @@
+
+{@a missing-translation}
+
+When a translation is missing, the build succeeds but generates a warning such as `Missing translation for message "foo"`.
```
Updated text.

```md
When a translation is missing, the build succeeds but generates a warning such as `Missing translation for message "{translation_text}"`.
```

josmar-crwdstffng committed:
docs: update link

Update link per review.

josmar-crwdstffng commented on aio/content/guide/i18n-common-locale-id.md:
```diff
@@ -0,0 +1,55 @@
+<header>Unicode locale identifiers</header>
+
+*   For a list of language codes, see [ISO 639-2][LocStandardsIso6392].
+*   IDs conform to the Unicode Common Locale Data Repository (CLDR).
```
Updated link since the site is insecure.

```md
[LocStandardsIso6392]: http://www.loc.gov/standards/iso639-2 "ISO 639-2 Registration Authority | Library of Congress"
```

josmar-crwdstffng reviewed (commented):
Acknowledged or implemented all edit review feedback.

josmar-crwdstffng commented on aio/content/guide/deprecations.md:
```diff
@@ -440,31 +460,34 @@ Given that this use case is relatively rare and special handling impacts applica
+For full rationale and discussion behind this deprecation, see [RFC: Internet Explorer 11 support deprecation and removal](https://github.com/angular/angular/issues/41840).
 
-*Note: IE11 will be supported in Angular v12 LTS releases through November 2022.*
+***Note***: *IE11 will be supported in Angular v12 LTS releases through November 2022.*
```
Updated text.

```md
**Note**: IE11 will be supported in Angular v12 LTS releases through November 2022.
```

josmar-crwdstffng commented on aio/content/guide/pipes.md:
```diff
@@ -223,20 +187,12 @@ Angular looks for changes to data-bound values in a [change detection](guide/glo
 
-* In the `flying-heroes.component.html (v1)` template, the `*ngFor` repeater displays the hero names.
-* Its companion component class `flying-heroes.component.ts (v1)` provides heroes, adds heroes into the array, and resets the array.
+*   In the `flying-heroes.component.html (v1)` template, the `*ngFor` repeater displays the hero names.
```
@TeriGlover The linking is not handled in this file.

mary-poppins commented:
You can preview 7d2a259 at https://pr43245-7d2a259.ngbuilds.io/.

josmar-crwdstffng committed:
docs: update content

Update content per review.

Co-authored-by: TeriGlover <teriglover@google.com>

josmar-crwdstffng committed:
docs: update punctuation

Update punctuation per review.

Co-authored-by: TeriGlover <teriglover@google.com>

josmar-crwdstffng committed:
docs: update content

Update content per review.

mary-poppins commented:
You can preview ab47d0f at https://pr43245-ab47d0f.ngbuilds.io/.

TeriGlover commented:
@googlebot I consent.

TeriGlover commented:
@googlebot I consent.

josmar-crwdstffng commented:
@googlebot I consent.

petebacondarwin commented:
@TeriGlover will need to consent to fix the CLA issue.
And the PR needs a rebase to resolve the conflicting files.

josmar-crwdstffng commented:
> @TeriGlover will need to consent to fix the CLA issue.
> And the PR needs a rebase to resolve the conflicting files.

@petebacondarwin 
The rebase was successful and @TeriGlover comment `@googlebot I consent.` multiple times.
What is the next step?

petebacondarwin commented:
The next step is to get a Googler to override the bot

mary-poppins commented:
You can preview 8b2d586 at https://pr43245-8b2d586.ngbuilds.io/.

josmar-crwdstffng committed:
docs: update review date

docs: Update review date for last content update.

mary-poppins commented:
You can preview 69183ba at https://pr43245-69183ba.ngbuilds.io/.

josmar-crwdstffng commented:
@gkalpak @alxhub @IgorMinar @JoostK

You were added as reviewers.  

The changes in the PR include the following.

*   Updated links in notes in `.ts` files
*   Separated `.md` file into multiple `.md` files
*   Added redirect
*   Updated left-nav toc

Please review the changes for technical accuracy.

mary-poppins commented:
You can preview 6050161 at https://pr43245-6050161.ngbuilds.io/.

aikithoughts reviewed (approved):
Reviewed-for: global-docs-approvers

mary-poppins commented:
You can preview b26ef3b at https://pr43245-b26ef3b.ngbuilds.io/.

mary-poppins commented:
You can preview 78659ba at https://pr43245-78659ba.ngbuilds.io/.

alxhub commented:
Hi @josmar-crwdstffng,

This PR needs to be rebased before it can land - thanks!

josmar-crwdstffng commented:
@alxhub
Rebased and pushed.  Please continue merge.
@josephperrott has flipped the `cla: yes` label multiple times.

mary-poppins commented:
You can preview 4eb47e5 at https://pr43245-4eb47e5.ngbuilds.io/.
